### DISCUSSION TITLE
新手请教在使用ioc注册newtownsoftJson后如何使用符合框架的Attribute

### DISCUSSION THREAD

**Discussion by DFFHSEVDH** [BYJRK/DotNet-Discussions#365] [C# 技术交流]
注册好NewtownsoftJson之后，我想使用JsonIgnore特性。忽然发现不知道该怎么下手，通过继承好像和直接用没什么区别了。

**Comment by BYJRK** [comment]
我没明白你的问题，你想在什么上使用 `JsonIgnore` 特性？以及你把 Newtonsoft.Json 注册为了什么服务？

**Reply by DFFHSEVDH** [reply]
我写了
``` c#
 public interface IJsonConvert
    {
        public string ToJson(object obj, bool indented = false);
        public T FromJson<T>(string json);
    }
```
这个接口，用来替换直接在文件中使用的Newtownsoft.Json。然后我发现想使用JsonIgnore特性时候就不会写了。还是说Json直接使用会更好一点。

**Reply by WHYBBE** [reply]
我大概明白你什么意思了（因为我之前也试图干过这样的事情）
我觉得直接使用 `Newtonsoft.Json` 的静态方法就好了，没必要再自己封装一层

可以看 `JsonIgnoreAttribute` 的代码：
```csharp
[AttributeUsage(AttributeTargets.Property | AttributeTargets.Field, AllowMultiple = false)]
public sealed class JsonIgnoreAttribute : Attribute
{
}
```
可以发现他是一个密封类没办法继承，如果你自己实现一个 `Attribute` 它是不认的

而且 JSON 序列化主流就两套（选一套就好），应该不会频繁变化，没必要去试图创建一套抽象去兼容两套实现（或者说去兼容 JSON 序列化以外的序列化方法）

**Reply by DFFHSEVDH** [reply]
多谢多谢，大概明白了。
